uaclient/apt.py

```diff
@@ -490,10 +495,33 @@ def is_installed(pkg: str) -> bool:
     return pkg in get_installed_packages()
 
 
-def get_installed_packages() -> List[str]:
+def generic_installed_packages() -> List[InstalledAptPackages]:
```

Comment:
@orndorffgrant I mean a function like this which will have all the package details from `apt list --installed` and thus there is no need for a separate function for versions or architecture(we might need it later, who knows e.g in the manifest file).